src/documents/signals/handlers.py

```diff
@@ -225,6 +225,28 @@ def set_tags(sender,
 @receiver(models.signals.post_delete, sender=Document)
 def cleanup_document_deletion(sender, instance, using, **kwargs):
     with FileLock(settings.MEDIA_LOCK):
+        if settings.TRASH_DIR:
+            counter = 0
+            old_filename = os.path.split(instance.source_path)[1]
+            (old_filebase, old_fileext) = os.path.splitext(old_filename)
+
+            while True:
```

Comment:
I think all this logic looks good to me, but I'd definitely like some commentary on this section to make it clearer why we're doing this filename recomposition. (I understand now, but it took me a second!)

Reply:
I've added a comment on the reason for the file-rename section.